stan/math/rev/core/chainablestack.hpp

```diff
@@ -11,6 +11,38 @@ class chainable_alloc;
 
 using ChainableStack = AutodiffStackSingleton<vari, chainable_alloc>;
 
+// Helper struct to access the underlying stack allocator.
+struct stack_mem_adapter {
+  template <typename T>
+  inline T* alloc_array(size_t n) const {
```

Comment:
There's no need to have an instance here.  Everything can be static.  So you can have

```
static inline void* alloc(size_t len) const {
  return ChainableStack::instance_->memalloc_.alloc(len);
}
```

and then usage is 

```
stack_mem_adapter::alloc(len);
```

rather than
```
stack_mem.alloc(len);
```

But you can also go even further and just make these simple inline functions.  Then usage reduces to 

```
stan::math::alloc(len);
```

which is perhaps a bit too exposed for such a simple function name.